src/main/groovy/life/qbic/utils/DataSetParser.groovy

```diff
@@ -0,0 +1,35 @@
+package life.qbic.utils
+
+/**
+ * A parser that takes an input data structure and tries to convert it to a certain type dataset
+ * of type T.
+ *
+ * The user of this interface can expect a valid dataset of type T if no exceptions have
+ * been thrown after parsing of data structure has been called.
+ *
+ * @since 1.7.0
+ */
+interface DataSetParser<T> {
+
+    /**
+     * Tries to parse and validate a data structure from a given path on the filesystem.
+     *
+     * The root path must be absolute!
+     *
+     * Given a dataset with the structure:
+     *
+     * - /SomePath/MyDataset
+     *      |- myFile.txt
+     *      |- anotherFile.txt
+     *
+     * Then /SomePath/MyDataset would be the root path.
+     *
+     * @param rootPath The root path of the dataset structure, represents the top level of the
+     * hierarchical data set structure.
+     * @return A successfully parsed and validated dataset of type T
+     * @throws DataParserException if the data structure cannot be parsed (unknown structure)
+     * @throws DataSetValidationException if the data structure is missing some required properties
+     */
+    T parseFrom(String rootPath) throws DataParserException, DataSetValidationException
```

Comment:
```suggestion
    T parseDirectory(String rootPath) throws DataParserException, DataSetValidationException
```
This makes it more explicit what we want the method to do. ParseFrom could also be from a file or name, an environment variable, ...
Since we already specified that the `rootPath` should be a string we can be explicit here.

Reply:
I oppose that `parse from a path` is pretty explicit :) I can change the parameter to `Path`. The rest is more a personal flavor.

Reply:
I think we should change the parameter then. If we want to provide a different method in the future a single string argument cannot be used otherwise.
e.g. `parseFrom(String multilinestring)`

Reply:
I don't think `parseFrom (Object path)` is not explicit enough. It was rather the concern of the parameter type being so open to be filled with different information (disregarding the parameter name)